### DISCUSSION TITLE
Migrating existing postal instance to a new server

### DISCUSSION THREAD

**Discussion by davids355** [postalserver/postal#2685] [Help with using Postal] [answered]
I would like to migrate an existing instance of postal to a new server - my goal is to retain the existing information regarding domains and credentials.

What I have done so far is - 

1. set up a new instance of postal on my new server
2. changed all DNS records to point postal FQDN to new server
3. I have a fresh instance of postal working at this point
4. Then I made a dummy organisation on the new server
5. Then I imported two databases - postal and postal-server-2, into the new server
6. I did notice that the new server had only postal and postal-server-1. So I imported databases accordingly.
7. When I go into the organisation on the new server I receive an error "we're sorry but something went wrong"

I am just wondering if this should work, if there is anything (or any databases) that I am missing and if its maybe the difference in database names that could have caused the issue?

**Comment by willpower232** [comment, marked as answer]
> Then I made a dummy organisation on the new server

It sounds very close however it looks like you fully set up the new server before moving the databases across (that is why you saw postal-server-1 and not postal-server-2).

Almost all of Postals configuration is in the databases so if you empty the new database server and then only add the contents from the old database server, that should do the trick.

**Comment by davids355** [comment]
Ah, just that small detail solved the problem! Thanks so much for taking the time to reply.

Just to clarify the entire process that I used for migrating from one server to another (Keeping the same domain), in case anyone else has the same problem and searches up this thread - 

1. I set up a new server with the same operating system as the current server (Ubuntu 20.04.6 TLS).
2. I followed the standard install guide here - https://docs.postalserver.io/install/installation/
3. I changed the DNS records to replace the old servers IP address with the new servers IP address (Otherwise I dont think Caddy would have been able to issue a LetsEncrypt certificate to the new installation).
4. I initially followed right the way through the installation guide AND logged in to the dashboard AND set up an organisation and an email server (But now I realise this step was incorrect, as per notes from WIllpower232 above).

When I imported data to postal and postal-server-2 from the backup taken from the old server, postal crashed when I tried to open the organisation.

Amending my procedure as per the above notes from Willpower232 I done the following - 

1. Copied sql backup file into the mariadb docker container - docker cp /backups/postal.sql docker-container-id:/postal.sql
2. Accessed mariadb docker container - docker exec -it postal-mariadb bash
3. connected to the mariadb - mariadb -uroot -p
4. dropped existing tables - drop table postal; drop table postal-server-2;
5. recreated both tables - CREATE DATABASE postal CHARSET utf8mb4 COLLATE utf8mb4_unicode_ci;
CREATE DATABASE `postal-server-2` CHARSET utf8mb4 COLLATE utf8mb4_unicode_ci;
6. Granted permissions for both tables - GRANT ALL ON `postal`.* TO `root`@`127.0.0.1` IDENTIFIED BY "db-password";
GRANT ALL ON `postal-server-2`.* TO `root`@`127.0.0.1` IDENTIFIED BY "db-password";
7. Imported sql backups to both tables - use postal; source /backups/postal.sql; use postal-server-2; source /backups/postal-server-2.sql;
8. Then i exited out of mariadb and the docker container and re-ran the postal initialize command - postal initialize
9. Then everything worked as expected :-)
